tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -1193,7 +1540,32 @@ namespace tuplex {
 
             if (symbol == "math.degrees")
                 return createMathToDegreesCall(builder, argsType, retType, args);
+            
+            if (symbol == "math.isnan")
+                return createMathIsNanCall(builder, argsType, retType, args);
+            
+            if (symbol == "math.isinf")
+                return createMathIsInfCall(builder, argsType, retType, args);
+                
+            if (symbol == "math.isclose") {
+                if (args.size() != 2 && args.size() != 3 && args.size() != 4) {
+                    std::string err = "math.isclose needs 2, 3, or 4 args; got " + std::to_string(args.size()) + " args\n";
+                    throw std::runtime_error(err);
+                }
+
+                assert(argsType.isTupleType());
+                assert(args.size() == argsType.parameters().size());
 
+                // check all argument types
+                std::vector<python::Type> input_types = argsType.parameters();
+                for(const auto& type : input_types) {
+                    if (type != python::Type::BOOLEAN && type != python::Type::I64 && type != python::Type::F64) {
+                        throw std::runtime_error("math.isclose argument must be a float, integer, or boolean");
```

Comment:
add to error message maybe what type the i-th argument is, so the user gets some more information.